proto/batch_plan.proto

```diff
@@ -30,6 +30,8 @@ message RowSeqScanNode {
   }
   // If along with `batch_limit`, `chunk_size` will be set.
   ChunkSize chunk_size = 6;
+  // The pushed down `batch_limit`. Max rows needed to return.
+  optional uint64 limit = 7;
```

Comment:
Is it possible also push offset into scan? I guess we can skip some `encoding/decoding` resource?

Reply:
In most cases, we can't push the offset to scan. Considering our scan has 4 partitions, it could cause an wrong answer if we push offset to each partition.

Reply:
`select * from t limit 10, 20`? We can push `10, 20` to all partitions? Anyway this is just an optimization, not a big problem.